frontend/packages/metal3-plugin/src/components/baremetal-hosts/BareMetalHostStatus.tsx

```diff
@@ -50,30 +50,67 @@ const BareMetalHostStatus: React.FC<BareMetalHostStatusProps> = ({
   nodeMaintenance,
 }) => {
   const statusTitle = title || status;
+  let hostStatus: JSX.Element;
+  const action = HOST_STATUS_ACTIONS[status]?.(host);
   switch (true) {
-    case status === HOST_STATUS_DISCOVERED:
-      return <AddDiscoveredHostButton host={host} />;
     case [NODE_STATUS_STARTING_MAINTENANCE, NODE_STATUS_UNDER_MAINTENANCE].includes(status):
-      return <MaintenancePopover title={statusTitle} nodeMaintenance={nodeMaintenance} />;
+      hostStatus = <MaintenancePopover title={statusTitle} nodeMaintenance={nodeMaintenance} />;
+      break;
     case [NODE_STATUS_STOPPING_MAINTENANCE, ...HOST_PROGRESS_STATES].includes(status):
-      return <ProgressStatus title={statusTitle}>{description}</ProgressStatus>;
+      hostStatus = (
+        <ProgressStatus title={statusTitle}>
+          {description}
+          {action}
+        </ProgressStatus>
+      );
+      break;
     case HOST_ERROR_STATES.includes(status):
-      return (
+      hostStatus = (
         <ErrorStatus title={statusTitle}>
           <p>{description}</p>
           <p>{getHostErrorMessage(host)}</p>
+          {action}
         </ErrorStatus>
       );
+      break;
     case HOST_SUCCESS_STATES.includes(status):
-      return <SuccessStatus title={statusTitle}>{description}</SuccessStatus>;
-    default:
-      return <Status status={status} title={statusTitle} />;
+      hostStatus = (
+        <SuccessStatus title={statusTitle}>
+          {description}
+          {action}
+        </SuccessStatus>
+      );
+      break;
+    case HOST_INFO_STATES.includes(status):
+      hostStatus = (
+        <InfoStatus title={statusTitle}>
+          {description}
+          {action}
+        </InfoStatus>
+      );
+      break;
+    default: {
+      const statusBody = <Status status={status} title={statusTitle} />;
+
+      hostStatus =
+        description || action ? (
+          <PopoverStatus title={statusTitle} statusBody={statusBody}>
+            {description}
+            {action}
+          </PopoverStatus>
+        ) : (
+          statusBody
+        );
+    }
   }
+
+  return hostStatus;
```

Comment:
I'd prefer returning from the cases instead of setting 'hostStatus' variable and breaking. No pressure, though :)

Reply:
fixed